### ISSUE TITLE
`brew uses` with `--include-optional` doesn't return used formula

### ISSUE THREAD

**Issue by echosa** [Homebrew/brew#1776]
# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is problem with Homebrew/brew and not specific formulae? If it's a formulae-specific problem please file this issue at https://github.com/Homebrew/homebrew-core/issues/new

### Bug reports:

When I run `brew uses` with `--include`, it doesn't find a formula that is actually used.

```
$ brew uses --installed --include-build --recursive unixodbc
homebrew/php/php70

$ brew uses --installed --include-build --include-optional --recursive unixodbc

$
```

Notice that unixodc is used by php70, but when I run the `uses` command with `--include-optional`, the command no longer shows php70 as using unixodbc.

`brew config`:

```
$ brew config
HOMEBREW_VERSION: 1.1.6-2-gfc11f633b2
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: fc11f633b29e6a2c9c4d49d5fd5452549b328193
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 0a6cce09b2ff3e50128d2e0d3887b12f5bf78368
Core tap last commit: 3 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.11.0 => /usr/local/bin/git
Perl: /usr/local/bin/perl => /usr/local/Cellar/perl/5.24.0_1/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.4.0/bin/ruby
Java: 1.8.0_65, 1.6.0_65-b14-468
macOS: 10.11.6-x86_64
Xcode: 8.1
CLT: 8.2.0.0.1.1480973914
X11: 2.7.9 => /opt/X11
```

`brew doctor` (I can't upgrade Xcode because I'm still on El Capitan):

```
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: Found Ruby file outside beeftornado/rmtree tap formula directory
(/usr/local/Homebrew/Library/Taps/beeftornado/homebrew-rmtree/Formula):
  /usr/local/Homebrew/Library/Taps/beeftornado/homebrew-rmtree/brew-rmtree.rb

Warning: Your Xcode (8.1) is outdated.
Please update to Xcode 8.2 (or delete it).
Xcode can be updated from the App Store.
```

**Comment by echosa** [comment]
@alyssais Thank you so much!

**Comment by alyssais** [comment]
@echosa looks like I introduced a bug and it had to be reverted but I'll get that fixed and back up soon :)

**Comment by echosa** [comment]
That's unfortunate. Keep me informed! Is there another ticket/PR for it I can watch?

**Comment by echosa** [comment]
@alyssais Just curious, any progress on this? If there's another PR or issue I should watch, please let me know. Not trying to rush it, just trying to stay informed. :-)

**Comment by alyssais** [comment]
Yes, sort of. The fix I have in mind also affects #1848, so I was waiting for that to be confirmed as a bug before I did it. I'll make sure to keep you updated here.

**Comment by echosa** [comment]
Great, thanks!
